trieste/acquisition/optimizer.py

```diff
@@ -197,68 +199,172 @@ def optimize_continuous(
-            opt_result = _perform_optimization(initial_points[i : i + 1], optimizer_args_local)
-            if opt_result.success:
-                successful_optimization = True
-
-                new_point = variable  # [1, D]
-                new_point_score = -opt_result.fun  # [1, 1]
-
-                if new_point_score > chosen_point_score:  # if found a better point then keep
-                    chosen_point = new_point  # [1, D]
-                    chosen_point_score = new_point_score  # [1, 1]
-
-        if not successful_optimization:  # if all optimizations failed then try from random start
-            for i in tf.range(num_recovery_runs):
-                random_start = space.sample(1)
-                if isinstance(space, TaggedProductSearchSpace):
-                    bounds = get_bounds_of_box_relaxation_around_point(space, random_start)
-                else:
-                    bounds = spo.Bounds(space.lower, space.upper)
-                optimizer_args_local["bounds"] = bounds
-                opt_result = _perform_optimization(random_start, optimizer_args_local)
-                if opt_result.success:
-                    chosen_point = variable  # [1, D]
-                    successful_optimization = True
-                    break
-            if not successful_optimization:  # return error if still failed
-                raise FailedOptimizationError(
-                    f"""
+        chosen_point, successful_optimization = _perform_parallel_continuous_optimization(
+            target_func,
+            space,
+            initial_points,
+            optimizer_args,  # run num_optimization_runs in parallel
+        )
+
+        if not successful_optimization:  # if all optimizations failed then try from random starts
+            random_points = space.sample(num_recovery_runs)  # [num_recovery_runs, D]
+            chosen_point, successful_optimization = _perform_parallel_continuous_optimization(
+                target_func, space, random_points, optimizer_args
+            )
+
+        if not successful_optimization:  # return error if still failed
+            raise FailedOptimizationError(
+                f"""
                     Acquisition function optimization failed,
                     even after {num_recovery_runs + num_optimization_runs} restarts.
                     """
-                )
-
-        return tf.convert_to_tensor(chosen_point)  # convert chosen point back from a variable
+            )
+        return chosen_point
 
     return optimize_continuous
 
 
+def _perform_parallel_continuous_optimization(
+    target_func: AcquisitionFunction,
+    space: SearchSpace,
+    starting_points: TensorType,
+    optimizer_args: dict[str, Any],
+) -> tuple[TensorType, bool]:
+    """
+    A function to perform parallel optimization of our acquisition functions
+    using Scipy. We perform L-BFGS-B starting from each of the locations contained
+    in `starting_points`.
```

Comment:
lets add that number of optimization runs is inferred from the shape of starting points